addons/mail/static/src/core/common/thread.js

```diff
@@ -302,7 +302,13 @@ export class Thread extends Component {
                 (this.props.order === "asc" &&
                     newerMessages &&
                     (loadNewer || thread.scrollTop !== "bottom"));
-            if (snapshot && messagesAtTop) {
+            if (this.scrollingToPresent) {
+                this.present.el?.scrollIntoView({
+                    behavior: "instant",
+                    block: "center",
+                });
+                this.scrollingToPresent = false;
```

Comment:
Similar to other remark, I don't like to add an extra condition here, we already have `thread.scrollTop` to control where we should go.